db/rdbms/migration/0002_migrate_descriptor_to_extended_descriptor.go

```diff
@@ -0,0 +1,285 @@
+
+	"github.com/facebookincubator/contest/cmds/plugins"
+	"github.com/facebookincubator/contest/pkg/pluginregistry"
+
+	"github.com/sirupsen/logrus"
+)
+
+const shardSize = uint64(50)
+
+// DescriptorMigration represents a migration which moves steps description in jobs tables from old to new
+// schema. The migration consists in the following:
+//
+//
+// In v0001, job.Request object was structured as follows:
+//
+// type Request struct {
+//		JobID         types.JobID
+//		JobName       string
+//		Requestor     string
+//		ServerID      string
+//		RequestTime   time.Time
+//		JobDescriptor string
+//		TestDescriptors string
+// }
+//
+// Having TestDescriptors as a field of request objects creates several issues, including:
+// * It's an abstraction leakage, the extended description of the steps might not be part
+//   of the initial request submmitted from outside (i.e. might not be a literal embedded
+//	 in the job descriptor)
+// * The job.Descriptor object does not contain the extended version of the test steps, which
+//   will make it difficult to handle resume, as we need to retrieve multiple objects
+//
+// Schmema v0002 introduces the concept of extended_descriptor, which is defined as follows:
+// type ExtendedDescriptor struct {
+//		JobDescriptor
+//		StepsDescriptors []test.StepsDescriptors
+// }
+//
+// We remove TestDescriptors from Request objects, and we store that information side-by-side with
+// JobDescriptor into an ExtendedDescriptor. We then store this ExtendedDescriptor in the jobs table
+// so that all the test information can be re-fetched simply by reading extended_descriptor field in
+// jobs table.
+
+type DescriptorMigration struct {
+	log *logrus.Entry
+}
+
+func ms(d time.Duration) float64 {
+	return float64(float64(d.Microseconds()) / 1000.0)
+}
+
+// fetchJobs fetches job requests based on limit and offset
+func fetchJobs(tx *sql.Tx, limit, offset uint64, log *logrus.Entry) ([]job.Request, error) {
+
+	log.Debugf("fetching shard limit: %d, offset: %d", limit, offset)
+	selectStatement := "select job_id, name, requestor, server_id, request_time, descriptor, teststeps from jobs limit ? offset ?"
+	log.Debugf("running query: %s", selectStatement)
+
+	start := time.Now()
+	rows, err := tx.Query(selectStatement, limit, offset)
+
+	elapsed := time.Since(start)
+	log.Debugf("select query executed in: %.3f ms", ms(elapsed))
+
+	if err != nil {
+		return nil, fmt.Errorf("could not get job request (limit %d, offset %d): %w", limit, offset, err)
+	}
+	defer func() {
+		_ = rows.Close()
+	}()
+
+	var jobs []job.Request
+
+	start = time.Now()
+	for rows.Next() {
+		job := job.Request{}
+		err := rows.Scan(
+			&job.JobID,
+			&job.JobName,
+			&job.Requestor,
+			&job.ServerID,
+			&job.RequestTime,
+			&job.JobDescriptor,
+			&job.TestDescriptors,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("could not scan job request (limit %d, offset %d): %w", limit, offset, err)
+		}
+		jobs = append(jobs, job)
+	}
+
```

Comment:
I was recently told one needs to check `rows.Err()`, to tell apart legit ends of  `rows.Next()` from error cases (https://golang.org/pkg/database/sql/#Rows)